gradle.properties

```diff
@@ -17,3 +17,8 @@ org.gradle.jvmargs=-Xmx1g -XX:MaxMetaspaceSize=512m -XX:+HeapDumpOnOutOfMemoryEr
 org.gradle.caching=true
 org.gradle.parallel=true
 
+# Use this flag so that when compiling against project dependencies
+# the jar is created and used instead of using src/main/classes.
+#
+# See https://docs.gradle.org/current/userguide/java_library_plugin.html#sub:java_library_known_issues_windows_performance
+systemProp.org.gradle.java.compile-classpath-packaging=true
```

Comment:
Why is this required?

Reply:
Ok, let me see. 

It begins with this very good question by @sbrannen:
> Why is org.opentest4j the only import with version information?

This was also a surprise to me because bnd gradle plugin was normally always able to determine the version to import a package because it would scan the classpath (essentially `project.sourceSet.main.compile`) and observe the package versions it found within the __jars__ that are there.

Remember, the bnd gradle plugin only adds metadata to the __jar__.

However, this was not working on this project! Why not?

I observed that this project uses the `java-library` gradle plugin which is an observably uncommon scenario as most gradle projects use the `java` plugin.

The distinctive difference here is that the `java` gradle plugin having dependencies of type `project` (e.g. `compile project(":foo")`) will result the project __jars__ being in `project.sourceSet.main.compile`. As I indicated earlier, this allowed bnd to observe the OSGi metadata for those dependent projects.

Now, the `java-library` plugin does _not_ by default cause a project dependency's jar to be built, AND when looking at the ``project.sourceSet.main.compile` will only contain the project output directory. Bnd does not emit metadata into the output directory and therefore there is no metadata to observer.

Setting the flag above to `true` tells the `java-library` plugin to revert to building the jar and using this in `project.sourceSet.main.compile`.

Reply:
Also, as I indicated in the comment, you can read more about this here https://docs.gradle.org/current/userguide/java_library_plugin.html#sub:java_library_known_issues_windows_performance

Reply:
Or better https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_classes_usage